src/main/scala/scala/compat/java8/collectionImpl/Stepper.scala

```diff
@@ -0,0 +1,545 @@
+package scala.compat.java8.collectionImpl
+
+import java.util.Spliterator
+
+/** A Stepper is a specialized collection that can step through its
```

Comment:
What's the value in exposing this as a collection type? Wouldn't it suffice to keep it an implementation detail and only expose spliterators and streams?